Let's compare the dataset that I have created with that that Xiangnan He did

In [27]:
# Original data
import numpy as np
import pandas as pd
import os

ratings = pd.read_csv(os.path.join(input_dir, input_fname),
    sep='::', engine='python',
    names=['userid', 'itemid', 'rating', 'timestamp'])

# python starts counting from 0
ratings['userid'] = ratings['userid'] - 1
ratings['itemid'] = ratings['itemid'] - 1

#itemsids are not continuous, which need to be for the embeddings lookup
item_map = dict(
    zip(sorted(ratings.itemid.unique()), range(len(ratings.itemid.unique())))
    )
ratings['itemid'] = ratings.itemid.replace(item_map)

# sort by userid and timestamp
ratings.sort_values(['userid','timestamp'], ascending=[True,True], inplace=True)
ratings.reset_index(inplace=True, drop=True)
ratings.head()

userid  itemid  rating  timestamp
0       0    2969       4  978300019
1       0    1178       5  978300055
2       0    1574       4  978300055
3       0     957       5  978300055
4       0    2147       3  978300103

In [29]:
df_xiangnan = pd.read_csv("Data_Xiangnan/ml-1m.test.rating", sep="\t",
    names=['userid', 'itemid', 'rating', 'timestamp'])

df_xiangnan.head()

userid  itemid  rating  timestamp
0       0      25       5  978824351
1       1     133       3  978300174
2       2     207       4  978298504
3       3     208       4  978294282
4       4     222       2  978246585

In [30]:
df_javier = pd.read_csv("Data_Javier/ml-1m.test.rating", sep="\t")
df_javier.head()

userid  itemid  rating  timestamp
0       0      47       5  978824351
1       1    1737       3  978300174
2       2    1900       4  978298504
3       3    1774       5  978294282
4       4     279       2  978246585

At firts sight one would say that my itemid 47 is Xiangnan He's itemid 25. Let's check out

In [50]:
df_xiangnan[df_xiangnan.itemid==25]

userid  itemid  rating  timestamp
0          0      25       5  978824351
5750    5750      25       2  958309656

In [51]:
df_javier[df_javier.itemid==47]

userid  itemid  rating  timestamp
0          0      47       5  978824351
2029    2029      47       3  977809337
4876    4876      47       2  962767153

These results seem inconsistent. Let's see what is the most recent review for userid 5750 in my dataset and let's also have a look to userid 5750's most recent reviews

In [53]:
df_javier[df_javier.userid==5750]

userid  itemid  rating  timestamp
5750    5750     950       4  958309656

In [55]:
ratings[ratings.userid==5750].tail(3)

userid  itemid  rating  timestamp
952148    5750    1961       3  958309656
952149    5750      47       2  958309656
952150    5750     950       4  958309656

We see that movies 47 and 950 were reviewed in the same second. In other words any of the 3 movies before could be in the testing dataset and would still be correct. Ok, the mapping seems consistent. This changes should not affect the recommendations. Let's double check with userid 2029.

In [56]:
ratings[ratings.userid==2029].tail(3)

userid  itemid  rating  timestamp
346522    2029    1908       3  977809337
346523    2029    3371       3  977809337
346524    2029      47       3  977809337

In [57]:
df_xiangnan[df_xiangnan.userid==2029]

userid  itemid  rating  timestamp
2029    2029    1047       3  977809337

We know that itemid 47 in my dataset corresponds to itemid 25 in Xiangnan He's. Therefore, Xiangnan He's itemid 1047 must be itemid 3371 or 1908 in my dataset. Let's find another user with that itemid as the most recent review and find out

In [58]:
df_xiangnan[df_xiangnan.itemid==1047]

userid  itemid  rating  timestamp
2029    2029    1047       3  977809337
5641    5641    1047       5  958925254

In [59]:
ratings[ratings.userid==5641].tail(3)

userid  itemid  rating  timestamp
934499    5641     121       1  958925187
934500    5641     957       2  958925187
934501    5641    1908       5  958925254

Therefore, itemid 1908 in my dataset is itemid 1047 in Xiangnan He's dataset. 

In conclusion, the mapping is consistent